File: solutions/python/flatten-array/1/flatten_array.py
from typing import Iterable, Any


def flatten(iterable: Iterable[Any]) -> list[Any]:
    result = []
    for x in iterable:
        if x is None:
            continue
        if isinstance(x, list):
            result.extend(flatten(x))
        else:
            result.append(x)
    return result

File: solutions/go/flatten-array/1/flatten_array.go
package flatten

func Flatten(a any) []any {
	switch a := a.(type) {
	case nil:
		return []any{}
	case []any:
		r := []any{}
		for _, x := range a {
			r = append(r, Flatten(x)...)
		}
		return r
	default:
		return []any{a}
	}
}
